ml-data-hub-plugin/src/main/groovy/com/marklogic/gradle/task/CleanJobsDataTask.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 MarkLogic Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.marklogic.gradle.task
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.marklogic.client.dataservices.ExecCaller
+import com.marklogic.client.dataservices.IOEndpoint
+import com.marklogic.client.io.JacksonHandle
+import com.marklogic.gradle.exception.RetainDurationRequiredException
+import com.marklogic.hub.MappingManager
+import com.marklogic.hub.scaffold.Scaffolding
+import org.gradle.api.tasks.TaskAction
+
+class CapturingErrorListener implements ExecCaller.BulkExecCaller.ErrorListener {
```

Comment:
All of this needs to shift into Java code; we want as little as possible in the Gradle task. All that the Gradle task should do is collect the retainDuration value and call a Java method. 

For the location in src/main/java - while we have a com.marklogic.hub.job package, jobs are really a part of the flow subsystem; i.e. jobs are not separate from flows (the only way to end up with a job is to run a flow; so really, the JobStatus class should move to hub.flow, but that's a separate matter). So let's do com.marklogic.hub.flow.JobManager and add a deleteJobs(retainDuration) method to it. No interface needed either, this isn't yet a public API.